text/0000-api-doc-frontpage-styleguide.md

```diff
@@ -0,0 +1,174 @@
+- Feature Name: api_doc_frontpage_styleguide
+- Start Date: 2016-07-25
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC is a companion to the [API doc guidelines proposal]
+(https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md).
+In this RFC, we focus on the format and style of the "front page" of an API.  The goal of this 
+RFC is to outline a clear presentation style for crate documentation that is helpful to new users 
+of a crate. These are the guidelines we'll be using for the rust-lang crates, but are also meant to
+help guide community crate authors.
+
+# Motivation
+
+A user's first experience with a crate is the front page of the crate's documentation. Taken
+together with the API documentation, the front page is a key piece of understanding a crate's
+purpose, how to use it, and its limitations.  Making sure this front page is clear will help users
+of your crate get the most out of it.
+
+# Detailed design
+
+The front page of the crate should cover four main points.  In this RFC, we cover each in turn.
+
+* [Introduction text to the crate](#introduction-text-to-the-crate) 
+* [First Example](#first-example) 
+* [Crate Capabilities](#crate-capabilities) 
+* [Source Examples](#source-examples)
+
+## Introduction text to the crate
+
+The first thing potential users of your crates will see is the introductory summary. This section is
+a good place to introduce the what and the why the crate. A good introduction is concise but also
+provides enough information for the reader to understand the purpose of the crate. In addition, a
+good introduction provides a learning path for the user as they read the rest of the documentation.
+
+Here's an example of a good introduction from the [log
+crate](https://doc.rust-lang.org/log/log/index.html):
+    
+> A lightweight logging facade.  
+>
+> A logging facade provides a single logging API that abstracts over the actual logging
+> implementation. Libraries can use the logging API provided by this crate, and the consumer
+> of those libraries can choose the logging framework that is most suitable for its use case.
+>
+> If no logging implementation is selected, the facade falls back to a "noop" implementation
+> that ignores all log messages. The overhead in this case is very small - just an integer load,
+> comparison and jump.  > > A log request consists of a target, a level, and a body. A target
+> is a string which defaults to the module path of the location of the log request, though that
+> default may be overridden. Logger implementations typically use the target to filter requests
+> based on some user configuration.
+
+There are several things that make this a good introduction section:
+    
+* A concise explanation of the crate's purpose.
+* The summary describes how the crate behaves in 3 short paragraphs.
+* Even if you don't want to read the rest of the `README`, you have enough to get started.
```

Comment:
I always find myself copying the description from the crate-level introduction to README. Something that we could work on abstracting.